plugin/path_regions.go

```diff
@@ -0,0 +1,141 @@
+package keystoneauth
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathListRegions(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "regions/?$",
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ListOperation: b.pathRegionList,
+		},
+	}
+}
+
+func pathRegions(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "regions/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"id": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "id",
+			},
+			"description": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "description",
+			},
+			"parent_region_id": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "parent_region_id",
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathRegionWrite,
+			logical.ReadOperation:   b.pathRegionRead,
+		},
+	}
+}
+
+func (b *backend) Region(s logical.Storage, n string) (*regionEntry, error) {
+	entry, err := s.Get("region/" + n)
```

Comment:
I would recommend to refactor error checks to:
```
if entry, err := s.Get("region/" + n); err != nil {
  return nil, err
}
// continue function
```
You are using this syntax already in some places. Let's keep it consequent.

Reply:
`entry` is scoped to `if` block but is require outside the block, line 55... so this is one way to do it...